src/mon/OSDMonitor.cc

```diff
@@ -12648,22 +12668,30 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
       goto reply;
     }
 
-    if (expected_num_objects > 0 &&
-	cct->_conf->osd_objectstore == "filestore" &&
-	cct->_conf->filestore_merge_threshold > 0) {
+    bool osd_all_bluestore = is_osd_all_bluestore();
```

Comment:
```suggestion
    set<int32_t> osds;
    osdmap.get_all_osds(osds);
    bool has_filestore_osd = std::any_of(osds.begin(), osds.end(), [this](int osd) {
      string type;
      if (!get_osd_objectstore_type(osd, &type)) {
        return type == "filestore";
      } else {
        return false;
      }
    });
```

in which:
1. no need to use an extra `checked_osds` to track the osds which have been checked, as `osdmap.get_all_osds()` returns a **set** which has no duplicated elements in it.
1. use `std::any_of()` for better readability.
1. avoid using negative conditions for better readability.

Reply:
Wonderful! looks up have a lot to learn.